scripts/host-check

```diff
@@ -1169,6 +1182,90 @@ def check_shmem_pages_max_size() -> CheckFuncReturn:
+    values.
+
+    :param module:
+        Name of the module
+
+    :param exp_param_vals:
+        Dictionary mapping parameter name to the expected value of the parameter
+
+    :return:
+        A dict containing parameters with unexpected values
+        If failed to check value for a parameter, returns None as the dict
+        value for the parameter
+    """
+    param_unexpected: Dict[str, Optional[str]] = {}
+
+    if not _is_module_loaded(
+        module.replace("-", "_")
+    ) and not _is_module_builtin(module):
+        # Skip module param check if module is not loaded
+        return param_unexpected
+    # sysfs filesystem requires the underscore version of the kernel module
+    # name
+    module = module.replace("-", "_")
+    all_params_str, _ = run_cmd(["modinfo", "--field", "parm", module])
+    for param, exp_val in exp_param_vals.items():
+        if not re.search(rf"^{param}:", all_params_str, re.MULTILINE):
+            # if 'param' is absent in modinfo output, it means either the
+            # parameter was introduced in a later kernel version, or the
+            # specific linux distribution does not support the parameter.
+            continue
+        try:
+            path = f"/sys/module/{module}/parameters/{param}"
+            with open(path, "r", encoding="utf-8") as f:
+                param_val = f.read().strip()
+            # (null) implies the parameter isn't set, and is thus the
+            # default value
+            if param_val not in (exp_val, "(null)"):
+                param_unexpected[param] = param_val
+        except FileNotFoundError:
+            param_unexpected[param] = None
+
+    return param_unexpected
+
+
+def check_module_non_default_params() -> CheckFuncReturn:
+    """
+    Checks the kernel modules have been loaded with default runtime parameters
+    (Checks parameters in MODULE_DEFAULT_PARAM)
+    XRd only uses default parameters, and that this check would need to change
+    if XRd required a non-default parameter to be set.
+    """
+    all_modules_str = ""
+    for module, param_defaults in MODULE_DEFAULT_PARAM.items():
+        module_str = ""
+        param_non_default = _module_param_compare(module, param_defaults)
+        for param, param_val in param_non_default.items():
+            if param_val is not None:
+                module_str += (
+                    f"\nThe default value for parameter {param} is "
+                    f"{param_defaults[param]}, but it is set to {param_val}"
+                )
+            else:
+                module_str += f"\nFailed to check value for parameter: {param}"
+        if module_str:
+            module_str = textwrap.indent(module_str, "   ")
+            all_modules_str += f"\nFor kernel module: {module}{module_str}"
+    if all_modules_str:
+        return (
+            CheckState.NEUTRAL,
+            "XRd has only been tested with default kernel module parameters."
```

Comment:
Change this to 'XRd has not been tested with these kernel module parameters (and then print details below).